src/crystal/scheduler.cr

```diff
@@ -0,0 +1,44 @@
+require "./event_loop"
+require "thread"
+
+class Thread
+  # :nodoc:
+  def scheduler
+    @scheduler ||= Crystal::Scheduler.new
```

Comment:
Currently the scheduler is global. In order to keep the current behavior this should be a class var.
I think there will be multiple schedulers with MT, so the accessing _the scheduler_ via the current thread is ok.

Reply:
Is there a downside to attach the scheduler on the thread now?